crates/payout_curve/src/lib.rs

```diff
@@ -0,0 +1,670 @@
+                    .to_u64()
+                    .expect("to be able to fit decimal into u64"),
+                outcome_payout: total_collateral,
+                extra_precision: 0,
+            },
+            PayoutPoint {
+                event_outcome: BTCUSD_MAX_PRICE
+                    .to_u64()
+                    .expect("to be able to fit decimal into u64"),
+                outcome_payout: total_collateral,
+                extra_precision: 0,
+            },
+        ),
+        // if coordinator is short, he gets 0 from his liquidation point to infinity
+        Direction::Short => (
+            PayoutPoint {
+                event_outcome: liquidation_price_upper_bound
+                    .to_u64()
+                    .expect("to be able to fit decimal into u64"),
+                outcome_payout: 0,
+                extra_precision: 0,
+            },
+            PayoutPoint {
+                event_outcome: BTCUSD_MAX_PRICE,
+                outcome_payout: 0,
+                extra_precision: 0,
+            },
+        ),
+    };
+
+    Ok((lower_range_lower, lower_range_upper))
+}
+
+/// Calculates lower and upper bounds for our payout curve, aka the liquidation prices
+fn calculate_payout_curve_bounds(
+    initial_price: Decimal,
+    leverage_trader: f32,
+    leverage_coordinator: f32,
+    coordinator_direction: Direction,
+) -> Result<(Decimal, Decimal)> {
+    let (liquidation_price_lower_bound, liquidation_price_upper_bound) = match coordinator_direction
+    {
+        Direction::Long => {
+            let leverage_short = Decimal::try_from(leverage_trader)?;
+            let liquidation_price_short =
+                calculate_short_liquidation_price(leverage_short, initial_price);
+
+            let leverage_long = Decimal::try_from(leverage_coordinator)?;
+            let liquidation_price_long =
+                calculate_long_liquidation_price(leverage_long, initial_price);
+            // if the coordinator is long, his lower bound is when he gets liquidated and the upper
+            // bound is when the trader gets liquidated
+            (liquidation_price_long, liquidation_price_short)
+        }
+        Direction::Short => {
+            let leverage_short = Decimal::try_from(leverage_coordinator)?;
+            let liquidation_price_short =
+                calculate_short_liquidation_price(leverage_short, initial_price);
+
+            let leverage_long = Decimal::try_from(leverage_trader)?;
+            let liquidation_price_long =
+                calculate_long_liquidation_price(leverage_long, initial_price);
+            // if the coordinator is short, his lower bound is when the trader gets liquidated and
+            // the upper bound is when the coordinator gets liquidated
+            (liquidation_price_long, liquidation_price_short)
+        }
+    };
+    Ok((liquidation_price_lower_bound, liquidation_price_upper_bound))
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
🙃 For ItchySats we used to have property-based tests which were quite useful because there are some properties that you want all your discretised payout curves to respect, for example:

- Total payout (coordinator + trader) is the same (or very close) for all payouts.
- Payout intervals have no gaps.
- Payout intervals are monotonically increasing (this one wasn't always true in ItchySats).

Reply:
Yes, I'll look into that because we get random overflow errors sometimes :/

Reply:
I wonder why they are so slow. Have you tried running with a different [config](https://altsysrq.github.io/proptest-book/proptest/tutorial/config.html) to reduce the number of test cases?

Reply:
I believe it's because we iterate over thousands of points. I might be able to optimize it by taking bigger steps, e.g. steps of 10

Reply:
Yeah, we definitely don't need so many steps when we want to discretise to a couple hundred CETs anyway.